testng-core/src/test/java/test/support/SafeRandoms.java

```diff
@@ -0,0 +1,23 @@
+package test.support;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+public final class SafeRandoms {
+
+  private static final Random random = new SecureRandom();
+
+  private SafeRandoms() {}
+
+  public static int nextInt(int bound) {
+    return nextInt(random, bound);
+  }
+
+  public static int nextInt(Random random, int bound) {
+    int value = 0;
+    while (value == 0) {
+      value = random.nextInt(bound);
+    }
+    return value;
+  }
```

Comment:
How about `1 + nextInt(bound - 1)` instead?

Reply:
I used this while loop based on the recommendation from the aibot that we are using for code reviews, wherein it pointed that using a `1` as constant would lead to the random being biased towards `1` whenever `nextInt()` returns 0. The refactored logic ensures that it tries to generate another ranom number if it finds a zero as the returned value.

Reply:
@krmahadevan , please check again. What I suggested was

```java
  public static int nextInt(Random random, int origin, int bound) {
    // require bound > origin
    return origin + random.nextInt(bound - origin);
  }
```

It is very different from what you had in https://github.com/testng-team/testng/pull/3100#discussion_r1534497691

Reply:
So for example I call `int value = SafeRandoms.nextInt(random, 100, 500)` and lets say `random.nextInt(500-100)` returned `0`, then the method would return back `100 + 0` which is `100`. 

Wouldn't that basically do the same thing if `nextInt()` call returned 0 (which the aibot was hinting at)? Instead of returning `1` we would return `origin` right?  Instead of affinity to `1` now it will be `100`. Is my understanding correct ?

Reply:
It is not the the same thing. Try simulating the output for nextInt(1, 2).

Reply:
@vlsi - Please check the below sample. The below test will always pass, which suggests that whenever `random.nextInt()` generates a `0` then our utility method will always return the origin, which is what the aibot comment was also about.

```java
import org.testng.annotations.Test;

import java.util.Random;

import static com.google.common.base.Preconditions.checkArgument;
import static org.testng.Assert.assertEquals;

public class RandomGenerationTest {

    private final FakeRandom random;

    public RandomGenerationTest() {
        random = new FakeRandom();
    }

    @Test(invocationCount = 10)
    public void testAffinityForRandoms() {
        int actual = nextInt(random, 10, 50);
        assertEquals(actual, 10, "The Random generated value should have always been 10");
    }

    public static int nextInt(Random random, int origin, int bound) {
        checkArgument(bound > origin, "Upper bound should be greater than origin");
        return origin + random.nextInt(bound - origin);
    }

    public static class FakeRandom extends Random {
        @Override
        public int nextInt(int ignored) {
            return 0;
        }
    }
}
```

Reply:
The only difference would be that, the affinity will now be based on the origin value that we pass, instead of it being a universal value of `1`. 

I am assuming that, that's the difference you are hinting at. Am I right @vlsi ?

Reply:
Random that always returns 0 is invalid

Reply:
Yes I know. That was just for demo purposes to show what I was trying to explain.

Reply:
Imagine nextInt(3). Imagine we call it 6 times, and it returns 0, 0, 1, 1, 2, 2.
Your suggestion was `if (==0) return 1`, and it would produce 1,1,1,1,2,2. Note how 1 appears twice as often as 2.

My suggestion was `offset+nextInt(bound-offset+1)` (apparently I missed -1, but, well, sure I meant it)

It means 1+nextInt(2) which yields 1,1,2,2,3,3. All the values are properly distributed.

Reply:
Yeah got it.
That was what I was cross checking when I mentioned 

> The only difference would be that, the affinity will now be based on the origin value that we pass, instead of it being a universal value of 1.

earlier.